etna/models/nn/tft_native/tft.py

```diff
@@ -0,0 +1,691 @@
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import List
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from typing_extensions import TypedDict
+
+from etna import SETTINGS
+from etna.distributions import BaseDistribution
+from etna.distributions import FloatDistribution
+from etna.distributions import IntDistribution
+
+if SETTINGS.torch_required:
+    import torch
+    import torch.nn as nn
+
+    from etna.models.base import DeepBaseModel
+    from etna.models.base import DeepBaseNet
+    from etna.models.nn.tft_native.layers import GateAddNorm
+    from etna.models.nn.tft_native.layers import StaticCovariateEncoder
+    from etna.models.nn.tft_native.layers import TemporalFusionDecoder
+    from etna.models.nn.tft_native.layers import VariableSelectionNetwork
+    from etna.models.nn.tft_native.loss import QuantileLoss
+
+
+class TFTNativeBatch(TypedDict):
+    """Batch specification for TFT."""
+
+    decoder_target: "torch.Tensor"
+    static_reals: Dict[str, "torch.Tensor"]
+    static_categoricals: Dict[str, "torch.Tensor"]
+    time_varying_categoricals_encoder: Dict[str, "torch.Tensor"]
+    time_varying_categoricals_decoder: Dict[str, "torch.Tensor"]
+    time_varying_reals_encoder: Dict[str, "torch.Tensor"]
+    time_varying_reals_decoder: Dict[str, "torch.Tensor"]
+
+
+class TFTNativeNet(DeepBaseNet):
+    """TFT based Lightning module."""
+
+    def __init__(
+        self,
+        encoder_length: int,
+        decoder_length: int,
+        n_heads: int,
+        num_layers: int,
+        dropout: float,
+        hidden_size: int,
+        lr: float,
+        static_categoricals: List,
+        static_reals: List,
+        time_varying_categoricals_encoder: List,
+        time_varying_categoricals_decoder: List,
+        time_varying_reals_encoder: List,
+        time_varying_reals_decoder: List,
+        categorical_feature_to_id: Dict,
+        loss: QuantileLoss,
+        optimizer_params: Optional[dict],
+    ) -> None:
+        """Init TFT.
+
+        Parameters
+        ----------
+        encoder_length:
+            encoder length
+        decoder_length:
+            decoder length
+        n_heads:
+            number of heads in Multi-Head Attention
+        num_layers:
+            number of layers in LSTM layer
+        dropout:
+            dropout rate in rnn layer
+        hidden_size:
+            size of the hidden state
+        lr:
+            learning rate
+        static_categoricals:
+            categorical features for the whole series, e.g. `segment`
```

Comment:
How are we going to control static features? In `TSDataset` we doens't have any flags about static features.

Reply:
What means "control"? User is responsible for his inputs